### PR TITLE
feat(web): improve UI/UX for settings pages

### PR DESCRIPTION
**Pull request by alextran1502**
This PR adds enhancements for the settings pages. 

- New accordion UI to give a visual clue that the setting is being opened
- Add an icon to each setting to help distinguish each of them more easily
- Add a search box to the admin settings since the list has grown off-screen, you can search using the set locale of the web app

### New UI
image

### Text search
image

### CODE REVIEW

File: web/src/routes/admin/system-settings/+page.svelte [hunk 1]
```diff
@@ -68,104 +90,128 @@
...
       subtitle: $t('admin.metadata_settings_description'),
       key: 'metadata',
+      icon: mdiDatabaseOutline,
     },
     {
       component: LibrarySettings,
       title: $t('admin.library_settings'),
       subtitle: $t('admin.library_settings_description'),
       key: 'external-library',
+      icon: mdiLibraryOutline,
```
Comment (ben-basten) [inline web/src/routes/admin/system-settings/+page.svelte:128]: I think this should match the icon in the sidebar.

```suggestion
      icon: mdiBookshelf,
```

File: web/src/lib/components/shared-components/settings/setting-accordion.svelte [hunk 2]
```diff
@@ -38,20 +40,28 @@
...
     on:click={() => (isOpen = !isOpen)}
     class="flex w-full place-items-center justify-between text-left"
   >
     <div>
-      <h2 class="font-medium text-immich-primary dark:text-immich-dark-primary">
-        {title}
-      </h2>
+      <div class="flex gap-2 place-items-center">
+        {#if icon}
+          <Icon path={icon} class="text-immich-primary dark:text-immich-dark-primary" size="24" />
```
Comment (ben-basten) [inline web/src/lib/components/shared-components/settings/setting-accordion.svelte:56]: Small change - hiding this icon from the screen reader.

```suggestion
          <Icon path={icon} class="text-immich-primary dark:text-immich-dark-primary" size="24" ariaHidden />
```

**Review by ben-basten — COMMENTED** [review]
This is a nice visual refresh! What do you think about moving the search bar out of the header, to spanning the width above the settings? This space is already particularly cramped in small screen sizes, and I think it would be cool to make this feature more visible and give users more space to see the text they have typed.

Screenshot 2024-09-13 at 07 18 23

File: web/src/lib/components/shared-components/settings/setting-accordion.svelte [hunk 3]
```diff
@@ -38,20 +40,28 @@
   });
 </script>
 
-<div class="border-b-[1px] border-gray-200 py-4 dark:border-gray-700" bind:this={accordionElement}>
+<div
+  class={`border rounded-2xl my-4 px-6 py-4 transition-all ${isOpen ? 'border-immich-primary/40 dark:border-immich-dark-primary/50 shadow-md' : 'dark:border-gray-800'}`}
```
Comment (jrasm91) [inline web/src/lib/components/shared-components/settings/setting-accordion.svelte:44]: Can you replace the es6 strings in favor of the svelte template syntax?

File: web/src/routes/admin/system-settings/+page.svelte [hunk 4]
```diff
@@ -68,104 +90,128 @@
...
       key: 'video-transcoding',
+      icon: mdiVideoOutline,
     },
   ];
+
+  let searchQuery = '';
+
+  $: filteredSettings = settings.filter(({ title, subtitle }) => {
+    const query = searchQuery.toLowerCase();
+    return title.toLowerCase().includes(query) || subtitle.toLowerCase().includes(query);
```
Comment (jrasm91) [inline web/src/routes/admin/system-settings/+page.svelte:213]: Can you include the actual config content on here as well? Basically index into it data.configs via the selected component.

**Review by jrasm91 — APPROVED** [review]

File: web/src/routes/admin/system-settings/+page.svelte (hunk 4 above)
Comment (alextran1502) [reply web/src/routes/admin/system-settings/+page.svelte:213]: Can you elaborate?

**Review by danieldietzler — APPROVED** [review]

File: web/src/routes/admin/system-settings/+page.svelte (hunk 4 above)
Comment (jrasm91) [reply web/src/routes/admin/system-settings/+page.svelte:213]: If I use Google oauth and search for "Google" would it return that result? 

If I search for aac and that is the code I'm using for transcoding would it return that section as well? 

I was just mentioning that you could also search against the current config for each second as well as the title and description.

**Pull request merged**